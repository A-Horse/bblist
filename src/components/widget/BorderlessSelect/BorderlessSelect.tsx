import React from 'react';
import { AppSelect } from '../AppSelect';
import { Props as SelectProps } from 'react-select/src/Select';
import { SelectOption } from '../../../typings/select.typing';
import './BorderlessSelect.scss';

function CustomSingleValue({ data, innerProps }) {
  return <div {...innerProps}>{data.label}</div>;
}

function CustomControl({ children, innerProps }) {
  return (
    <div {...innerProps} style={{ position: 'relative', cursor: 'pointer' }}>
      {children}
    </div>
  );
}

function CustomIndicator({}) {
  return null;
}

export function BorderLessSelector(
  props: SelectProps<SelectOption> & { width: number }
) {
  return (
    <div className="BorderLessSelector" style={{ width: props.width }}>
      <AppSelect
        {...props}
        styles={{
          indicatorsContainer: () => ({
            display: 'none'
          })
        }}
        components={{
          SingleValue: CustomSingleValue,
          Control: CustomControl,
          DropdownIndicator: CustomIndicator
        }}
      />
    </div>
  );
}
