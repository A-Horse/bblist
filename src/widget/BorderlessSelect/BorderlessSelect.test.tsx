import React from 'react';
import { BorderLessSelector } from './BorderlessSelect';
import renderer from 'react-test-renderer';

test('BorderLessSelector', () => {
  const assigneeSelector = renderer
    .create(<BorderLessSelector width={100} options={[]} />)
    .toJSON();
  expect(assigneeSelector).toMatchSnapshot();
});
