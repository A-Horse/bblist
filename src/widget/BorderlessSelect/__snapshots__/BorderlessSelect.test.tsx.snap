// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BorderLessSelector 1`] = `
<div
  className="BorderLessSelector"
  style={
    Object {
      "width": 100,
    }
  }
>
  <div
    className="AppSelect css-2b097c-container"
    onKeyDown={[Function]}
  >
    <div
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "position": "relative",
        }
      }
    >
      <div
        className="App-select__value-container css-1hwfws3"
      >
        <div
          className="App-select__placeholder css-1wa3eu0-placeholder"
        >
          Select...
        </div>
        <div
          className="css-w8afj7-Input"
        >
          <div
            className="App-select__input"
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-autocomplete="list"
              autoCapitalize="none"
              autoComplete="off"
              autoCorrect="off"
              disabled={false}
              id="react-select-2-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              spellCheck="false"
              style={
                Object {
                  "background": 0,
                  "border": 0,
                  "boxSizing": "content-box",
                  "color": "inherit",
                  "fontSize": "inherit",
                  "label": "input",
                  "opacity": 1,
                  "outline": 0,
                  "padding": 0,
                  "width": "1px",
                }
              }
              tabIndex="0"
              type="text"
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </div>
      </div>
      <div
        className="App-select__indicators css-9y64sj-IndicatorsContainer"
      >
        <span
          className="App-select__indicator-separator css-1okebmr-indicatorSeparator"
        />
      </div>
    </div>
  </div>
</div>
`;
